### ISSUE TITLE
Refactor dsverifier.cpp

### ISSUE THREAD

**Issue by ssvlab** [ssvlab/dsverifier#114]
- Review dsverifier.cpp methods
- Refactor according to design patterns
- Split dsverifier.cpp in other subclasses and methods
- Apply CPPLint
- Documentation: add comments for each method/class
Main goals:
(1) create C++ classes for representing digital-system and state-space objects;
(2) instantiate those classes from the main procedure of DSVerifier.

**Comment by feliperodri** [comment]
@lucasccordeiro, are we still going to use the DSVerifier's GUI? Do we really need to keep this code around?

**Comment by lucasccordeiro** [comment]
Yes, sure :-) We must still keep it running. May I ask you to try it again?
